packages/backend-auth/src/access_builder.test.ts

```diff
@@ -0,0 +1,57 @@
+import { describe, it, mock } from 'node:test';
+import { allowAccessBuilder } from './access_builder.js';
+import {
+  ConstructContainer,
+  ConstructFactoryGetInstanceProps,
+  ResourceAccessAcceptorFactory,
+  ResourceProvider,
+} from '@aws-amplify/plugin-types';
+import assert from 'node:assert';
+
+void describe('allowAccessBuilder', () => {
+  const resourceAccessAcceptorMock = mock.fn();
+
+  const getResourceAccessAcceptorMock = mock.fn(
+    // allows us to get proper typing on the mock args
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    (_: string) => resourceAccessAcceptorMock
```

Comment:
this should be solvable if you provide typing for function, 
see example here https://github.com/aws-amplify/amplify-backend/blob/9ad867e17d216b2292fb4f942a84f728e2471a5f/packages/client-config/src/client-config-writer/client_config_writer.test.ts#L18 .